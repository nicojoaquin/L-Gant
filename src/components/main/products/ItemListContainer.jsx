import React, {useState, useEffect} from 'react'
import ItemList from "./Itemlist"


const ItemListContainer = () => {

  const [items, setItems] = useState([])
  
  const products = [
    {id: 1, codeName:"sastreroGris", name: "Pantalon sastrero gris", price: 11000, category: "pantalones", img: "/assets/products/pantalon_gris.jpg" },
    {id: 2, codeName:"camisaBlanca", name: "Camisa blanca lisa", price: 7000, category: "camisas", img: "/assets/products/camisa_blanca.jpg"},
    {id: 3, codeName:"sacoNegro", name: "Saco negro liso", price: 22000, category: "sacos", img: "/assets/products/saco_negro.jpg" },
    {id: 4, codeName:"trajeRayado", name: "Traje rayado azul", price: 55000, category: "trajes", img: "/assets/products/traje_rayado_azul.jpg"}
  ]

  useEffect( () => {
     
    //Creamos una promesa que carga los productos en 2 segundos.
    const getItems = new Promise((resolve, reject) => {

      let ok = true

      ok ? setTimeout(() => resolve(setItems(products)), 2000) : 
      reject("Algo ha salido mal")
      
    })
  
    getItems.then((result) => {
     return result
    })
    .catch((err) => {
      console.error(err);
    })

  }, [])

  
  return (
    <>

    <h2>Productos</h2>
    <div className = "item-container">
      <ItemList items= {items} /> {/* Le pasamos el estado de los productos ya cambiado(agregados). */}
    </div>

    </>
  )
}

export default ItemListContainer;
