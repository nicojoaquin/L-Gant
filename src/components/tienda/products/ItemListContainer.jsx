import React, { useState, useEffect } from 'react'
import { useParams } from 'react-router';
import useProducts from '../../../hooks/useProducts';
import Category from './Category';
import ItemList from "./Itemlist"
// import { collection, onSnapshot } from '@firebase/firestore';
// import db from "../../../firebase-config"


const ItemListContainer = () => {

  const {catId} = useParams()
  const {data, loader} = useProducts();
  const [items, setItems] = useState([])

  useEffect(() => {
    if(catId === "all"){
      setItems(data) 
    }
    else {
      setItems(data.filter(dt => dt.category === catId))  
    }
  },[data, catId])


  return (

    /*Cuando termina de cargar, aparecen los productos.*/
    <div className = "container">

      {
        loader ?        
          <svg 
          className = "cssload-spin-box loader"
          style = {{marginTop: 400}}>
          </svg> 
        :     
        <div>
          <h2 className = "products-title">Productos</h2>
          <br />
          <hr />
          <div className = "item__container">  
            <Category />
            <ItemList products = {items} />           
          </div>
        </div>  
      }

    </div>

  )

}

export default ItemListContainer;