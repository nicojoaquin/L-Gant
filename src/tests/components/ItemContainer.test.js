import React from 'react';
import {shallow} from 'enzyme';
import ItemListContainer from '../../components/main/products/ItemListContainer'


describe('Prueba en <ItemListContainer />', () => {
  
  const buttonText = 'Agregar'
  const wrapper = shallow(<ItemListContainer buttonText = {buttonText} />);
  
  test('debe mostrar el componente correctamente', () => {

    
    expect( wrapper ).toMatchSnapshot();


  })

  test('debe tener un texto en el boton y la clase debe ser "button"', () => {
    
    const button = wrapper.find('button');
    expect( button.text().trim()).toBe(buttonText);
    expect(button.prop('className').includes('button')).toBe(true)


  })
    

})


