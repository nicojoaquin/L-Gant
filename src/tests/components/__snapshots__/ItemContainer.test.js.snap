// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <ItemListContainer /> debe mostrar el componente correctamente 1`] = `
<div
  className="container"
>
  <Finder
    products={Array []}
  />
  <div
    className="item__container"
  >
    <div
      className="cssload-spin-box"
    />
    <button
      className="button"
      disabled={false}
      onClick={[Function]}
    >
      Agregar
    </button>
    <Itemlist
      products={Array []}
    />
  </div>
</div>
`;
